src/Components/TopicPage.jsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { GetSingleTopic, getArticles } from "../Utility";
+import { Link } from "react-router-dom";
+
+function TopicPage() {
```

Comment:
It looks like the all articles and articles by topics pages are very similar. With that being the case, this could be acheived in the all articles page instead of having to creat a new component.